specifyweb/frontend/js_src/lib/components/QueryBuilder/Wrapped.tsx

```diff
@@ -144,12 +144,11 @@ export function QueryBuilder({
   );
 
   const isResourceModified = useIsModified(queryResource);
-  const saveRequired =
-    isResourceModified || (state.saveRequired && !isEmbedded);
-  const saveRequiredForEmbeded = isResourceModified || state.saveRequired;
+  const saveRequired = isResourceModified || state.saveRequired;
+  const promptToSave = !isEmbedded || saveRequired;
```

Comment:
careful, this should be an `&&`, not `||`.

I would also reorder the two conditions to make it more logical (you would prompt the user if "save is required as long as they are not in an embedded query" rather than if "the user is not in an embedded query, or and so happens they also need to save changed")

Thus:
```
const promptToSave = saveRequired && !isEmbedded;
```